### PR TITLE
chore: update initializer logic to perform checks on override files

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
This update ensures that we check if the overrides.js file exist before trying to execute override logic, so that it doens't throw all the time, and we can now throw if there is a runtime error.

E2E run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/12371/workflows/33547174-f9dc-415c-9e47-b8d918e2bfc9

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sobolk — COMMENTED** [review]
lgtm, please add tests.

File: packages/amplify-provider-awscloudformation/src/initializer.ts [hunk 1]
```diff
@@ -92,26 +92,41 @@ export const run = async (context: $TSContext): Promise<void> => {
...
-      }
+      overrideFilePath = path.join(backendDir, 'awscloudformation', 'build', 'override.js');
+      projectInitialized = true;
     } catch (e) {
-      // TODO: this always throws, and likely needs re-written
-      printer.debug(`Unable to apply auth role overrides: ${e.message}`);
+      // project not initialized
+    }
+    // when a new environment is created, init is called, and we must apply overrides for the auth/unauth roles
+    // not sure why we only do this for auth...
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/initializer.ts:105]: This code is not related to the `auth` category specifically. The `auth` and `unauth` role names are created on `init` for legacy implementation reasons. I think this comment adds more confusion than anything.

```suggestion
```

**Review by jhockett — APPROVED** [review]
